SecretSanta/test/SecretSanta.web.UITests/UserPageTests.cs

```diff
@@ -0,0 +1,192 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Mime;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using SecretSanta.web.UITests.WebPages;
+using SecretSanta.web.UITests.WebPages.UserPages;
+
+namespace SecretSanta.web.UITests
+{
+    [TestClass]
+    public abstract class UserPageTests
+    {
+        /* Because this URL is hardcoded, Changes to the port the web project runs on will break all of these tests unless this string is updated */
+        private const string RootUrl = "https://localhost:44331/";
+        public TestContext TestContext { get; set; }
+
+        protected IWebDriver Driver { get; set; }
+
+      
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            if (TestContext.CurrentTestOutcome == UnitTestOutcome.Failed)
+            {
+                SaveScreenshotOfFailure();
+            }
+
+            Driver.Quit();
+            Driver.Dispose();
+        }
+
+        private void SaveScreenshotOfFailure()
+        {
+            try
+            {
+                var projectDirectory = Directory.GetCurrentDirectory().Split(@"\bin").First();
+                var outputDirectory = Path.Join(projectDirectory, "FailedTests");
+
+                if (!Directory.Exists(outputDirectory))
+                {
+                    Directory.CreateDirectory(outputDirectory);
+                }
+
+                var outputFilename = Path.Join(outputDirectory, $"{TestContext.TestName}.png");
+                var screenshot = ((ITakesScreenshot) Driver).GetScreenshot();
```

Comment:
This works because the current driver supports screenshots, but what happens when the Driver doesn't implement ITakesScreenshot? Better to type check rather than simply cast.
```suggestion
                if (Driver is ITakesScreenshot takesScreenshot)
                {
                    var screenshot = takesScrrenshot.GetScreenshot();
```